store/sql_post_store.go

```diff
@@ -156,6 +156,82 @@ func (s SqlPostStore) Update(newPost *model.Post, oldPost *model.Post) StoreChan
 	return storeChannel
 }
 
+const (
+	// Set HasReactions = true if and only if the post has reactions, update UpdateAt only if HasReactions changes
+	UPDATE_POST_IS_PINNED = `UPDATE
+			Posts
+		SET
+			UpdateAt = :UpdateAt,
+			IsPinned = :IsPinned
+		WHERE
+			Id = :PostId`
+)
+func (s SqlPostStore) Pin(postId string) StoreChannel {
+	storeChannel := make(StoreChannel)
+
+	go func() {
+		result := StoreResult{}
+
+		if transaction, err := s.GetMaster().Begin(); err != nil {
+			result.Err = model.NewLocAppError("SqlPostStore.Pin", "store.sql_post.pin.begin.app_error", nil, err.Error())
+		} else {
+			_, err := transaction.Exec(UPDATE_POST_IS_PINNED, map[string]interface{}{"PostId": postId, "IsPinned": 1, "UpdateAt": model.GetMillis()})
+
+			if err != nil {
+				transaction.Rollback()
+				result.Err = model.NewLocAppError("SqlPostStore.Pin", "store.sql_post.pin.save.app_error", nil, err.Error())
+			} else {
+				if err := transaction.Commit(); err != nil {
+					// don't need to rollback here since the transaction is already closed
+					result.Err = model.NewLocAppError("SqlPostStore.Pin", "store.sql_post.pin.commit.app_error", nil, err.Error())
+				}
+			}
+
+			if result.Err == nil {
+				result.Data = postId
```

Comment:
Obviously this line throws an error as `Data` is not expecting a string. So, should I call SqlPostStore's Get to retrieve the updated post?